x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/utils.ts

```diff
@@ -91,3 +92,45 @@ export const getGapBetweenRuns = ({
 };
 
 export const makeFloatString = (num: number): string => Number(num).toFixed(2);
+
+/**
+ * Given a BulkResponse this will return an aggregation based on the errors if any exist
+ * from the BulkResponse. Errors are aggregated on the reason as the unique key.
+ *
+ * Example would be:
+ * {
+ *   'Parse Error': {
+ *      count: 100,
+ *      statusCode: 400,
+ *   },
+ *   'Internal server error': {
+ *       count: 3,
+ *       statusCode: 500,
+ *   }
+ * }
+ * If this does not return any errors then you will get an empty object like so: {}
+ * @param response The bulk response to aggregate based on the error message
+ * @param ignoreStatusCodes Optional array of status codes to ignore when creating aggregate error messages
+ * @returns The aggregated example as shown above.
+ */
+export const errorAggregator = (
+  response: BulkResponse,
+  ignoreStatusCodes: number[] = []
```

Comment:
Huh, TIL that typescript will infer this as optional given the default value

Reply:
Yeah, really since I use the array within my code I shouldn't be defaulting this at all as that leads to common errors and if the real code isn't using defaults and only the tests are using defaults it needs to require the array to be passed in.

To follow my own advice, using default parameters that have no use case in the code base === 'bad' and can cause more permutations and bugs later on.

I am going to make this required and change my tests.